Search Interval

stecychirinda opened the pull request:
### Description:
This pull request introduces a waiting state that periodically searches for possible matches. When no matches are found, it displays a "Searching for a buddy". The implementation uses setTimeout to search for matches at 30-second intervals for a total of 1 minute. If no matches are found after the timer runs out, the state changes to false and displays a "No buddy found" message.
Changes:

- Enter a waiting state and periodically search for possible matches
- Display a "Searching for a buddy" message 
- Check if the match array is empty and enter a searching state if it is
- Use setTimeout to search for matches at 30-second intervals for 1 minute
- Change state to false and display "No buddy found" when the timer runs out

Not Included in this PR:
Displaying a timer (will be implemented in a later PR)
### Milestones:
This pull request works towards Technical Challenge 2.
### Resources:
[Timers API Documentation](https://l.meta.ai/?u=https%3A%2F%2Fnodejs.org%2Fdocs%2Flatest%2Fapi%2Ftimers.html&h=AT1DDsci6hnFabWWGqUBlEXhYEAYHed_mghnFktpam_uBAMQAsJI3NzIbWF2V33EvzxYTqLey4a3Sw-juf1yzC__s-lsW22ZckNy7YRZEJinPm6R9e-2gFYo3JiOYljpJIs0_Ru5QZItcnUMTekswerADv8)
[Window clearTimeout API Documentation](https://developer.mozilla.org/en-US/docs/Web/API/Window/clearTimeout)
[Inspirational YouTube Video](https://l.meta.ai/?u=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D2I7BWkD-tGg&h=AT1DDsci6hnFabWWGqUBlEXhYEAYHed_mghnFktpam_uBAMQAsJI3NzIbWF2V33EvzxYTqLey4a3Sw-juf1yzC__s-lsW22ZckNy7YRZEJinPm6R9e-2gFYo3JiOYljpJIs0_Ru5QZItcnUMTekswerADv8)
### Test Plan:
The code was tested to ensure that it correctly displays the "No buddy found" message when there are no matches. A video of the test is attached.
[Searching Demo](https://pxl.cl/7J6KK)
Testing Details:

- Tested the waiting state with an empty match array
- Verified that the searching message is displayed 
- Verified that the state changes to false and displays "No buddy found" after the timer runs out

Tested for edge cases, including:

- No matches found after 1 minute

jzysak25 commented on Frontend/my-vite-app/src/Components/Homepage.jsx:
```diff
@@ -34,7 +34,36 @@ const homePage = ({ profile }) => {
+              navigate("/buddy", { state: { match: response.data.matched } });
+            }
+
+            if (timer <= 60000) {
```
no magic number

jzysak25 commented on Frontend/my-vite-app/src/Components/Homepage.jsx:
```diff
@@ -34,7 +34,36 @@ const homePage = ({ profile }) => {
+                state: { match: response.data.matched },
+              });
+            }
+          } catch (error) {
```
Here, you are catching the error and returning an error message. 
The issue here is:
1. You are returning a string, and it is consistent with ```return navigate("/buddy", {
                state: { match: response.data.matched },
              });```
2. You are not assigning the error message anywhere, meaning the error will be swallowed.

stecychirinda commented:
Thank you for the feedback.
I have implemented the suggested changes in my latest commit where I ensured that error messages are properly returned and not swallowed by sending the errors to the console.
I also refactored the code to store numbers in constant variables, eliminating magic numbers.
